support-workers/src/typescript/lambdas/createPaymentMethodLambda.ts

```diff
@@ -0,0 +1,194 @@
+import { combinedAddressLine } from '../model/address';
+import type {
+	DirectDebitPaymentFields,
+	PaymentFields,
+	PayPalPaymentFields,
+	StripePaymentFields,
+} from '../model/paymentFields';
+import type {
+	DirectDebitPaymentMethod,
+	PaymentMethod,
+	PayPalPaymentMethod,
+	StripePaymentMethod,
+} from '../model/paymentMethod';
+import { stageFromEnvironment } from '../model/stage';
+import type {
+	CreatePaymentMethodState,
+	CreateSalesforceContactState,
+	User,
+	WrappedState,
+} from '../model/stateSchemas';
+import {
+	createPaymentMethodStateSchema,
+	wrapperSchemaForState,
+} from '../model/stateSchemas';
+import { ServiceHandler } from '../services/config';
+import { getPayPalConfig, PayPalService } from '../services/payPal';
+import { getStripeConfig, StripeService } from '../services/stripe';
+import { getIfDefined } from '../util/nullAndUndefined';
+
+const stage = stageFromEnvironment();
+const stripeServiceHandler = new ServiceHandler(stage, async (stage) => {
+	const config = await getStripeConfig(stage);
+	return new StripeService(config);
+});
+const paypalServiceHandler = new ServiceHandler(stage, async (stage) => {
+	const config = await getPayPalConfig(stage);
+	return new PayPalService(config);
+});
+
+export const handler = async (
+	state: WrappedState<CreatePaymentMethodState>,
+) => {
+	console.info(`Input is ${JSON.stringify(state)}`);
+	const createPaymentMethodState = wrapperSchemaForState(
+		createPaymentMethodStateSchema,
+	).parse(state).state;
+	return createSalesforceContactState(
+		createPaymentMethodState,
+		await createPaymentMethod(
+			createPaymentMethodState.paymentFields,
+			createPaymentMethodState.user,
+		),
+	);
+};
+
+export function createPaymentMethod(
+	paymentFields: PaymentFields,
+	user: User,
+): Promise<PaymentMethod> {
+	switch (paymentFields.paymentType) {
+		case 'Stripe':
+			return createStripePaymentMethod(user.isTestUser, paymentFields);
+		case 'PayPal':
+			return createPayPalPaymentMethod(user.isTestUser, paymentFields);
+		case 'DirectDebit':
+			return createDirectDebitPaymentMethod(paymentFields, user);
+		case 'Existing':
+			return Promise.reject(
+				new Error(
+					'Existing payment methods should never make their way to this lambda',
+				),
+			);
+	}
+	return Promise.reject(new Error('Unknown payment method type'));
+}
+
+export function createSalesforceContactState(
+	inputState: CreatePaymentMethodState,
+	paymentMethod: PaymentMethod,
+): WrappedState<CreateSalesforceContactState> {
+	const outputState: CreateSalesforceContactState = {
+		...inputState,
+		paymentMethod,
+	};
+
+	return {
+		state: outputState,
+		error: null,
+		requestInfo: {
+			testUser: false,
```

Comment:
if you're hard coding to false, have you checked that none of the logic in the state machine uses this field?
(I assume the scala lambdas would use the one from the user?)

Reply:
https://github.com/guardian/support-frontend/blob/f2bbb0a87cf906276a7a17e12a36a5e5a0d1a0c6/cdk/lib/support-workers.ts#L174